components/org.wso2.carbon.identity.sts.passive.ui/src/main/java/org/wso2/carbon/identity/sts/passive/ui/PassiveSTS.java

```diff
@@ -613,6 +619,36 @@ private void handleAuthenticationRequest(HttpServletRequest request, HttpServlet
             throws IOException, ServletException, PassiveSTSException {
 
         Map paramMap = request.getParameterMap();
+        String tenantDomain = resolveTenantDomain(paramMap);
+        SessionDTO sessionDTO = buildSessionDTO(paramMap, tenantDomain, request.getQueryString());
+
+        String sessionDataKey = UUIDGenerator.generateUUID();
+        addSessionDataToCache(sessionDataKey, sessionDTO);
+
+        sendToAuthenticationFramework(request, response, sessionDataKey, sessionDTO, tenantDomain);
+    }
+
+    private String resolveTenantDomain(Map paramMap) {
+
+        String tenantDomain = null;
+        if (IdentityTenantUtil.isTenantQualifiedUrlsEnabled()) {
+            tenantDomain = IdentityTenantUtil.getTenantDomainFromContext();
```

Comment:
In tenant qualified mode we should return this value. 
We shouldn't fall back to query params?

@malithie  WDYT?

Reply:
addressed